tests/langchain/test_langchain_tracer.py

```diff
@@ -188,12 +189,17 @@ def test_llm_internal_exception():
         run_id=run_id,
         name="test_llm",
     )
-    callback._run_span_mapping = {}
-    with pytest.raises(
-        Exception,
-        match=f"Span for run_id {run_id} not found.",
-    ):
-        callback.on_llm_end(LLMResult(generations=[[{"text": "generated text"}]]), run_id=run_id)
+    try:
+        with pytest.raises(
+            Exception,
+            match="Span for run_id dummy not found.",
+        ):
+            callback.on_llm_end(LLMResult(generations=[[{"text": "generated"}]]), run_id="dummy")
+    finally:
+        callback.flush()
```

Comment:
```suggestion
    callback.flush()
```

is there a reason we need to call `flush` in finally?

Reply:
When the `on_llm_end` call does not raise any exception, the `pytest.raises` validation fails and `callback.flush` won't be executed, which results in the span context leakage.

Reply:
I see, can we make `MlflowLangchainTracer` a context manger and call `flush` in `__exit__`?

Reply:
Unfortunately not - the tracer should be a subclass of LangChain's base callback class, which is not a context manager. Also it needs to be passed as a callback object to their API like `invoke()` method ([ref](https://python.langchain.com/docs/how_to/callbacks_runtime/)).

Reply:
Can we add a method like this to `MlflowLangchainTracer`?

```python
    @contextlib.contextmanager
    def context(self):
        """
        Should be used only for testing purposes.
        """
        try:
            yield
        finally:
            self.flush()
```

Reply:
Not a blocker.

Reply:
Hmm we could but not sure if it's good way to add test-only code in the callback... It is an user-facing code (e.g., users may customize the logic like [this](https://mlflow.org/docs/latest/llms/langchain/autologging.html#customize-callback)). Let me leave it now as we have the pytest fixture as guardrail. When we need to do that in more places than the tracer unit test, we should add some utility.